src/proxy.ts

```diff
@@ -54,9 +58,18 @@ function computeCommunalApi3ReaderProxyV1Address(chainId: ethers.BigNumberish, d
 }
 
 function computeDappId(dappAlias: string, chainId: ethers.BigNumberish) {
-  if (!DAPPS.some((dapp) => dapp.alias === dappAlias)) {
+  const dApp = DAPPS.find((dapp) => Object.hasOwn(dapp.aliases, dappAlias));
+  if (!dApp) {
     throw new Error(`dApp with alias ${dappAlias} not registered to the package`);
   }
+  const chainAlias = CHAINS.find((chain) => chain.id === chainId)?.alias;
```

Comment:
You shouldn't be strict about this stuff in here, you're breaking the non-strict mode of the CLI command

Reply:
I added it for consistency 🤔 Shouldn't we then also remove the following?
```
  const dApp = DAPPS.find((dapp) => Object.hasOwn(dapp.aliases, dappAlias));
  if (!dApp) {
    throw new Error(`dApp with alias ${dappAlias} not registered to the package`);
  }
```

For now, removed all of those checks, but can revert in case I'm missing something.

Reply:
I agree on removing that too. Since we're doing a major change, the function can be renamed as `unsafeComputeDappId()` to mitigate potential issues because of the lack of checks.

Reply:
Huh https://github.com/api3dao/data-feed-reader-example/blob/31902685bc20e192069d5072393e925796366568/scripts/deploy-dapp-specific-api3readerproxyv1-unsafe.js#L11

Reply:
I was afraid for a moment that I messed up something with the rename. Do you want me to fire up a PR in data-feed-reader-example which uses this function from contracts?

Reply:
Nah, it's just that it looks like I was deliberate with the dApp check